TypeCobol.Analysis/Cfg/ControlFlowGraphBuilder.cs

```diff
@@ -986,67 +986,83 @@ private Node ResolveProcedure(Node node, SectionNode sectionNode, SymbolReferenc
+        /// <param name="performCallRelation">The Perform Call relation.</param>
+        private void ResolvePendingPERFORMProcedure(Tuple<PerformProcedure, SectionNode, BasicBlockForNodeGroup> perform, List<Tuple<PerformProcedure, SectionNode, BasicBlockForNodeGroup>> clonedPerforms,
+            PerformCallRelation performCallRelation)
         {
             PerformProcedure p = perform.Item1;
             SectionNode sectionNode = perform.Item2;
             BasicBlockForNodeGroup group = perform.Item3;
 
-            SymbolReference procedureReference = p.CodeElement.Procedure;
-            SymbolReference throughProcedureReference = p.CodeElement.ThroughProcedure;
-
-            Node procedureNode = ResolveProcedure(p, sectionNode, procedureReference);
-            if (procedureNode == null)
+            PerformTarget performTarget = GetPerformTarget(p, sectionNode);
+            if (performTarget == null)
                 return;
-
-            Procedure procedure = _nodeToProcedure[procedureNode];
             var clonedBlocksIndexMap = new Dictionary<int, int>();
-            if (throughProcedureReference != null)
-            {
-                Node throughProcedureNode = ResolveProcedure(p, sectionNode, throughProcedureReference);
-                if (throughProcedureNode == null)
-                    return;
-
-                Procedure throughProcedure = _nodeToProcedure[throughProcedureNode];
-                if (procedure.Number > throughProcedure.Number)
-                {
-                    // the second procedure name is declared before the first one.
-                    this.Cfg.AddWrongOrderPerformThru(p, procedureNode, throughProcedureNode);
-                    return;
-                }
-
-                //Accumulate sentences located between the two procedures
-                List<Sentence> sentences = new List<Sentence>();
-                int currentProcedureNumber = procedure.Number;
-                while (currentProcedureNumber <= throughProcedure.Number)
-                {
-                    var currentProcedure = this.CurrentProgramCfgBuilder.AllProcedures[currentProcedureNumber];
-                    currentProcedure.AccumulateSentencesThrough(sentences, throughProcedure, out var lastProcedure);
-                    currentProcedureNumber = lastProcedure.Number + 1;
-                }
-
-                //Store sentences into the group
-                StoreSentences(sentences);
-            }
-            else
-            {
-                //Store all sentences from the procedure into the group
-                StoreSentences(procedure);
-            }
-
+            StoreSentences(performTarget);
             //And finally relocate the Graph.
             RelocateBasicBlockForNodeGroupGraph(p, group, clonedBlocksIndexMap);
 
-            void StoreSentences(IEnumerable<Sentence> sentences)
+            void StoreSentences(PerformTarget target)
```

Comment:
Parameter is not used, remove it please.

Reply:
Done in 35d0734